### DISCUSSION TITLE
Is it possible to introduce a covariate into training/inference?

### DISCUSSION THREAD

**Discussion by paul-reiners** [MIC-DKFZ/nnUNet#655] [Q&A] [answered]
I'm using nnUNet to segment MRI scans of babies brains.  The structure of babies' brains changes from month to month quite radically.  We have been training separate models for separate age groups, such as 1-month-old, 2-months-old, and so on.  The problem is that we don't have a lot of images for each age group.

Is it possible to introduce a covariate into the training/inference that is an integer representing the age in months of the subject?  That way we could train on all of our images simultaneously.  This would certainly reduce training time and increase the accuracy of the model.

In addition to the T1 and T2 volumes, could a third volume be used which would be a constant gray-scale, where the darkness of the gray-scale is a multiple of the age in months?

**Comment by FabianIsensee** [comment, marked as answer]
Hi,
why not just train with all subjects without giving the age? That should work flawlessly.
Adding a greyscale image as additional input will not help because this will be destroyed during normalization (mean subtraction)
Best,
Fabian

**Comment by paulwrightwork** [comment]
I have a similar query. My use case is segmentation of ischaemic lesions on CT brain scans from acute stroke patients. It is a tricky segmentation, even for an expert human, and model performance reflects this with a wide spread in Dice. One issue is that the appearance of lesion changes qualitatively over time. I was hoping including the time from onset to scan in training might help the model account for this and improve the results.

My naive approach would be to generate a volume with all voxels having the time value as the OP considered. With CT, I believe absolute values are used to clip the voxel values instead of mean and SD as in MRI, but maybe the mean subtraction is a separate step. Is there any way to get this to work?

I'm a machine learning noob but understand the more correct approach would be to customize the code to inject these scalars at a later stage in training. I have no idea how to go about this and expect it would be labour-intensive but if anyone in this community wants to offer any suggestions, I would be grateful.
